blocks/map/map.js

```diff
@@ -0,0 +1,93 @@
+import {
+  div, iframe, button, p, horizontalRule,
+} from '../../scripts/dom-helpers.js';
+
+function getMapFrame(url) {
+  const mapFrame = iframe({
+    id: 'map-frame',
+    class: 'map-frame-aspect-ratio',
+    role: 'application',
+    src: url,
+    sandbox: 'allow-scripts allow-same-origin allow-forms allow-popups allow-popups-to-escape-sandbox',
+    allow: 'autoplay; geolocation; fullscreen;',
+    tabindex: '0',
+    loading: 'lazy',
+    title: 'Esri locations map',
+  });
+
+  return mapFrame;
+}
+
+function toggleFullscreen() {
+  const mapWrapper = document.querySelector('#frame-wrapper');
+  const mapFrame = document.querySelector('#map-frame');
+
+  const minimizeButton = document.querySelector('.return-btn');
+
+  if (window.screenTop && window.screenY) {
+    mapWrapper.classList.toggle('is-fullscreen');
+    mapFrame.classList.toggle('map-frame-aspect-ratio');
+    minimizeButton.classList.toggle('btn-vis');
+  } else {
+    mapWrapper.classList.toggle('is-fullscreen');
+    mapFrame.classList.toggle('map-frame-aspect-ratio');
+    minimizeButton.classList.toggle('btn-vis');
+  }
+}
+
+export default async function decorate(block) {
+  const blockParams = block.querySelectorAll('p');
+
+  const mapLink = blockParams[0].innerText;
+  const textParameter = blockParams[1].innerText;
+  block.textContent = '';
+
+  const fullscreenButton = button(
+    {
+      class: 'btn btn-white',
+      label: 'Enter full screen',
+    },
+  );
+
+  fullscreenButton.innerHTML = 'Launch map full screen';
```

Comment:
It is a good practice to only use `.textContent` instead of `.innerHTML` when possible.

```suggestion
  fullscreenButton.textContent = 'Launch map full screen';
```